.github/workflows/build.yml

```diff
@@ -0,0 +1,254 @@
+name: build
+
+on:
+  push:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 23 * * *'
+
+env:
+  # default user for pushing
+  DOCKER_USER: ${{ secrets.DOCKERHUB_USERNAME }}
+  # default token (password) for pushing
+  DOCKER_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+  # default prefix for all pushed images
+  IMAGE_PRE: dev-
```

Comment:
I tend towards keeping `dev-` repos. I think best would be to automatically push to them if not on master. It makes feature branches easier to handle. Otherwise I'll remove the option or set empty.

What do you think?

Reply:
What happens if some outsider forks their repo? Will their fork be built and pushed to our dockerhub or does is only build branches inside the Moonvision/moonbox-docker repo?

Reply:
I read up on it before. A that push doesn't trigger on PRs. The action for that is pull_request: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories

Confirmed: https://github.com/MoonVision/moonbox-docker/pull/2

If another fork runs, their CI will fail for permissions: https://github.com/ahirner/moonbox-docker/actions/runs/654920234

Reply:
To me it looks like PR #2 manages to push the docker image to the registry.

Reply:
You are absolutely right. I blinded over the `15 successful checks` ... Puzzling. I will push until fixes and comment when fixed.

Reply:
My prior test was wrong. It didn't contain a new commit. This test does and didn't trigger: https://github.com/MoonVision/moonbox-docker/pull/3

Reply:
> I tend towards keeping `dev-` repos. I think best would be to automatically push to them if not on master. It makes feature branches easier to handle. Otherwise I'll remove the option or set empty.

What would be the usecase for the dev- tags? Doesn't seem like a good idea to reference them from anywhere given how unstable they are.

Reply:
I want to avoid avoid regressions in downstream builds for features/tests but still avoid building all locally (compute, time). 
Of course `dev-` repos are not mutually exclusive to any non-default branch but their tags are stable during a PR or performance test.

I think an alternative solution to my problem would be removing `latest` tags if not on master.

Reply:
> tags are stable during a PR or performance test.

But they aren't stable in that context either. As soon as there are two branches open you are randomly referring to the images of either branch.

Reply:
I agree it doesn't have the best behaviour. Only `ver_tag` are stable no matter the repo (`dev-` or not).
What do you suggest for the use case?
My alternative: 
> I think an alternative solution to my problem would be removing latest tags if not on master.